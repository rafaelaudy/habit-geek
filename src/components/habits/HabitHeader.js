import React from "react";
import { useTranslation } from "react-i18next";
import { isToday } from "@rafael.audy/habit-geek-utils/utils/dateUtils";

const HabitHeader = ({ isReadOnly, week }) => {
  const { t } = useTranslation();
  const days = t("date-days-short", { returnObjects: true });

  const cellComponents = days.map((day, index) => {
    const todayClass = isToday(week, index)
      ? "habit-row-frequency__cell--today-header"
      : "";
    return (
      <h5
        key={`header-cell-${index}`}
        className={`rounded-top habit-row-frequency__cell ${todayClass}`}
      >
        {day}
      </h5>
    );
  });

  return (
    <div className="habit-row__header">
      <div className="habit-row__title"></div>
      <div className="habit-row-frequency">{cellComponents}</div>
      {isReadOnly ? null : <div className="icon-button"></div>}
    </div>
  );
};

export default HabitHeader;
