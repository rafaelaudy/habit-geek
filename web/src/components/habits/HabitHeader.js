import React from "react";
import { useTranslation } from "react-i18next";
import { isToday } from "@habit-geek/shared/utils/dateUtils";

const HabitHeader = ({ isReadOnly, week }) => {
  const { t } = useTranslation();
  const days = t("date-days-short", { returnObjects: true });

  const cellComponents = days.map((day, index) => {
    const todayClass = isToday(week, index) ? "habit__cell--today-header" : "";
    return (
      <h5
        key={`header-cell-${index}`}
        className={`rounded-top habit__cell ${todayClass}`}
      >
        {day}
      </h5>
    );
  });

  return (
    <div className="habit__header-row">
      <div className="habit__cell-habit-container"></div>
      <div className="habit__cell-frequency-container">{cellComponents}</div>
      {isReadOnly ? null : <div className="habit__cell-action-container"></div>}
    </div>
  );
};

export default HabitHeader;
