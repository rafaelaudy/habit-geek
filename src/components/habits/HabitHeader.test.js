import React from "react";
import { shallow } from "enzyme";
import HabitHeader from "./HabitHeader";
import { isToday } from "@rafael.audy/habit-geek-utils/utils/dateUtils";

jest.mock("@rafael.audy/habit-geek-utils/utils/dateUtils", () => ({
  isToday: jest.fn().mockReturnValue(false)
}));

describe("HabitHeader component", () => {
  it("Renders static elements", () => {
    const habitHeader = shallow(<HabitHeader />);
    expect(habitHeader.find(".icon-button").length).toBe(1);
    expect(habitHeader).toMatchSnapshot();
  });

  it("Hides header placeholder when on readonly mode", () => {
    const habitHeader = shallow(<HabitHeader isReadOnly={true} />);
    expect(habitHeader.find(".icon-button").length).toBe(0);
  });

  it("Shows border around current date", () => {
    isToday.mockReturnValueOnce(true);
    const habitHeader = shallow(<HabitHeader />);
    expect(
      habitHeader.find(".habit-row-frequency__cell--today-header").length
    ).toBe(1);
  });
});
