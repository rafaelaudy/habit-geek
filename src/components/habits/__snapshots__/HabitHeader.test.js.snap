// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HabitHeader component Renders static elements 1`] = `
<div
  className="habit-row__header"
>
  <div
    className="habit-row__title"
  />
  <div
    className="habit-row-frequency"
  >
    <h5
      className="rounded-top habit-row-frequency__cell "
      key="header-cell-0"
    >
      Mon
    </h5>
    <h5
      className="rounded-top habit-row-frequency__cell "
      key="header-cell-1"
    >
      Tue
    </h5>
    <h5
      className="rounded-top habit-row-frequency__cell "
      key="header-cell-2"
    >
      Wed
    </h5>
    <h5
      className="rounded-top habit-row-frequency__cell "
      key="header-cell-3"
    >
      Thu
    </h5>
    <h5
      className="rounded-top habit-row-frequency__cell "
      key="header-cell-4"
    >
      Fri
    </h5>
    <h5
      className="rounded-top habit-row-frequency__cell "
      key="header-cell-5"
    >
      Sat
    </h5>
    <h5
      className="rounded-top habit-row-frequency__cell "
      key="header-cell-6"
    >
      Sun
    </h5>
  </div>
  <div
    className="icon-button"
  />
</div>
`;
